### PR TITLE
[Metrics v2] join tests

### LINKED ISSUE #41899: QP changes to allow for joins

### PR DESCRIPTION
**Pull request by snoe**
Fixes #41899
PR to enable a disabled test and exercise more join work.

### CODE REVIEW

File: src/metabase/query_processor/middleware/resolve_joined_fields.clj [hunk 1]
```diff
@@ -50,7 +50,7 @@
       (let [explicit-joins (remove :fk-field-id joins)]
         (if (= (count explicit-joins) 1)
           (recur field {:joins explicit-joins} clause)
-          (let [{:keys [_id name]} (lib.metadata/table (qp.store/metadata-provider) table-id)]
+          (let [{:keys [name]} (lib.metadata/table (qp.store/metadata-provider) table-id)]
```
Comment (snoe) [inline src/metabase/query_processor/middleware/resolve_joined_fields.clj:53]: snake case hating map threw here

File: src/metabase/lib/card.cljc [hunk 2]
```diff
@@ -162,5 +162,5 @@
                         (assoc metric-query :lib/metadata (:lib/metadata query))
                         -1
                         (lib.util/query-stage metric-query -1)
-                        options)))
+                        (dissoc options :unique-name-fn))))
```
Comment (snoe) [inline src/metabase/lib/card.cljc:165]: If unique-name-fn gets triggered in here, then the rest of returned columns will be thrown off.
I was wondering if this should still use `card-metadata-columns` with `dataset-query` and `results-metadata` replaced.  I think that affects whether named or id references are used.

Comment (metamben) [reply src/metabase/lib/card.cljc:165]: How will the rest of the returned columns be thrown off?

Comment (snoe) [reply src/metabase/lib/card.cljc:165]: Unsure, I was getting a `PRODUCT_ID_2` in the second join condition in this query. But I can't reproduce it now

image

**Review by metamben — APPROVED** [review]

**Pull request merged**